numpy/core/src/common/get_attr_string.h

```diff
@@ -62,7 +62,7 @@ maybe_get_attr(PyObject *obj, char *name)
     /* Attribute referenced by (char *)name */
     if (tp->tp_getattr != NULL) {
         res = (*tp->tp_getattr)(obj, name);
-        if (res == NULL) {
+        if (res == NULL && PyErr_ExceptionMatches(PyExc_AttributeError)) {
             PyErr_Clear();
```

Comment:
The comments on line 45-48 aren't really accurate any more

Reply:
what would I write instead? What is there right now is
> Stripped down version of PyObject_GetAttrString,  avoids lookups for None, tuple, and List objects, and doesn't create a PyErr since this code ignores it.
>
> This can be much faster then PyObject_GetAttrString where exceptions are not used by caller.

It still does not create a PyErr (directly, that is), but if `obj` raises something in either `__getattr__` or `__getattribute__` that is not an `AttributeError`, it gets passed through.

Also, I don't understand the second part because `PyObject_GetAttrString` may be referenced by `tp->tp_getattr` (if that is set – which it normally is not, so `tp->tp_getattro` and `PyObject_GenericGetAttr` are used). My understanding of python's C-API is limited though, so I may be wrong.

Reply:
I went ahead and updated the comment myself